tests/runtime/service_generation/signature_parsing/test_parsers.py

```diff
@@ -102,6 +126,59 @@ def _run(self, foo):
     assert get_argument_types(_run)["foo"] == sample_lib.data_model.SampleInputType
 
 
+def test_get_args_with_no_annotation():
+    """Check that we get arguments with known arg type supplied"""
+
+    def _run(input_1="hello world"):
+        pass
+
+    assert get_argument_types(_run)["input_1"] == str
+
+    def _run_with_docstring(input_1):
+        """
+        Args:
+            input_1: str
+                Optional str input
+        """
+        pass
+
+    assert get_argument_types(_run_with_docstring)["input_1"] == Optional[str]
+
+    def _run_with_known_dm_type(sample_input_type):
+        pass
+
+    assert (
+        get_argument_types(_run_with_known_dm_type)["sample_input_type"]
+        == sample_lib.data_model.SampleInputType
```

Comment:
wait... how does this actually work?

I'm a bit confused on where we configure that the name `sample_input_type` -> `dm.SampleInputType`. Is this from `_get_dm_type_from_name`?

Reply:
Yes. We do `sample_input_type` converted snake to camel case `SampleInputType` and then get_dm_type_from_name . Gabe magic

Reply:
🪄🪄🪄